script.js

```diff
@@ -1,12 +1,104 @@
-//You can edit ALL of the code here
+const rootElem = document.getElementById("root");
+const dropdownEl = document.createElement("select");
+const searchInput = document.createElement("input"); // Creates an input element
+const searchStats = document.createElement("p"); // Counts the number of episodes displayed
+const searchContainer = document.createElement("div");
+const allShows = document.createElement("div"); // A container which displays the episodes
+let totalEpisodes; // Variable for the total number of episodes
+
+// Adds classes to the elements
+searchContainer.classList.add("search-container");
+searchInput.classList.add("search-input");
+searchStats.classList.add("search-stats");
+allShows.classList.add("all-shows");
+
+// Appends the elements
+searchContainer.appendChild(dropdownEl);
+searchContainer.appendChild(searchInput);
+searchContainer.appendChild(searchStats);
+rootElem.appendChild(searchContainer);
+rootElem.appendChild(allShows);
+
+// This event listener goes to the show which is selected in the drop down
+dropdownEl.addEventListener("change", (event) => {
+  document.location = `index.html#${event.target.value}`;
+});
+
+// This function when the page is loaded
 function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+  fetch("https://api.tvmaze.com/shows/82/episodes")
+    .then((response) => response.json())
+    .then((data) => fetchedEpisodes(data))
+    .catch((error) => console.log(error));
+}
+
+// This function fetches all the episodes and filters them if needed
+function fetchedEpisodes(data) {
+  let episodeList = data;
+  totalEpisodes = data.length; // This is used in line 54 for displaying the total episodes count
```

Comment:
don't reference line numbers in comments (line numbers change often as you code so this can become out of date quickly)